util/dvsim/LintCfg.py

```diff
@@ -0,0 +1,221 @@
+        '''
+        return LintCfg(flow_cfg_file, proj_root, args)
+
+    def gen_results_summary(self):
+        '''
+        Gathers the aggregated results from all sub configs
+        '''
+
+        # Generate results table for runs. Note that we build a simple md and
+        # a marked up md version here in parallel.
+        log.info("Create summary of lint results")
+
+        results_str = "## Lint Summary Results\n\n"
+        results_str += "### " + self.timestamp_long + "\n\n"
+
+        header = [
+            "Name", "Tool Warnings", "Tool Errors", "Lint Warnings",
+            "Lint Errors"
+        ]
+        colalign = ("center", ) * len(header)
+        table = [header]
+
+        num_result_rows = 0
+        for cfg in self.cfgs:
+
+            results_page = cfg.results_server_dir + '/results.html'
+            results_page_url = results_page.replace(
+                cfg.results_server_prefix, cfg.results_server_url_prefix)
+            name_with_link = "[" + cfg.name.upper(
+            ) + "](" + results_page_url + ")"
+            table.append([
+                name_with_link,
+                str(len(cfg.result_summary["warnings"])) + " W",
+                str(len(cfg.result_summary["errors"])) + " E",
+                str(len(cfg.result_summary["lint_warnings"])) + " W",
+                str(len(cfg.result_summary["lint_errors"])) + " E"
+            ])
+
+            num_result_rows += 1
+
+        if num_result_rows:
+            self.results_summary_md = results_str + tabulate(
+                table, headers="firstrow", tablefmt="pipe",
+                colalign=colalign) + "\n"
+        else:
+            self.results_summary_md = results_str + "\nNo results to display.\n"
+
+        print(self.results_summary_md)
+
+        # Return only the tables
+        return self.results_summary_md
+
+    def _gen_results(self):
+        # '''
+        # The function is called after the regression has completed. It looks
+        # for a regr_results.hjson file with aggregated results from the lint run.
+        # The hjson needs to have the following (potentially empty) fields
+        #
+        # {
+        #   tool: ""
+        #   errors: []
+        #   warnings: []
+        #   lint_errors: []
+        #   lint_warning: []
+        #   lint_infos: []
+        # }
+        #
+        # where each entry is a string representing a lint message. This allows
+        # to reuse the same LintCfg class with different tools since just the
+        # parsing script that transforms the tool output into the hjson above
+        # needs to be adapted.
+        #
+        # note that if this is a master config, the results will
+        # be generated using the _gen_results_summary function
+        # '''
+
+        # Generate results table for runs.
+        # Note that we build a simple md and a marked up md version
+        # here in parallel.
+        results_str = "## " + self.name.upper() + " Lint Results\n\n"
```

Comment:
Set `self.results_title` in `__post_init__` and use it here to match heading with the title:
```
    def __post_init__(self):
        # Run some post init checks
        super().__post_init__()
        self.results_title = self.name.upper() + "Lint Results"
```
This is used by the base class (FlowCfg) method to set the title of the results page (will be indicated as tooltip with you hover on the tab).

Reply:
done.